lib/src/core/verifier/policy/opa/opa.go

```diff
@@ -0,0 +1,78 @@
+package main
+
+import "C"
+
+import (
+	"context"
+	"encoding/json"
+	"strings"
+
+	"github.com/open-policy-agent/opa/rego"
+	"github.com/open-policy-agent/opa/storage/inmem"
+)
+
+//export evaluateGo
+func evaluateGo(policy string, data string, input string) *C.char {
+	// Deserialize the message in json format
+	input_map := make(map[string]interface{})
+	err := json.Unmarshal([]byte(input), &input_map)
+	if err != nil {
+		return C.CString("Error:: " + err.Error())
+	}
+
+	data_map := make(map[string]interface{})
+	err2 := json.Unmarshal([]byte(data), &data_map)
+	if err2 != nil {
+		// Handle error.
+		return C.CString("Error:: " + err.Error())
+	}
+	// Manually create the storage layer. inmem.NewFromObject returns an
+	// in-memory store containing the supplied data.
+	store := inmem.NewFromObject(data_map)
+
+	// Construct a Rego object that can be prepared or evaluated.
+	r := rego.New(
+		rego.Query("input;data.policy"),
+		rego.Module("demo.rego", policy),
+		rego.Store(store),
+	)
+
+	// Create a prepared query that can be evaluated.
+	ctx := context.Background()
+	query, err := r.PrepareForEval(ctx)
+	if err != nil {
+		return C.CString("Error:: " + err.Error())
+	}
+
+	// Make opa query
+	rs, err := query.Eval(ctx, rego.EvalInput(input_map))
+	if err != nil {
+		return C.CString("Error:: " + err.Error())
+	}
+
+	// Transform the processed decision into the format rust hopes for
+	inputOPA := rs[0].Expressions[0].Value.(map[string]interface{})
+	dataOPA := rs[0].Expressions[1].Value.(map[string]interface{})
```

Comment:
Can't these type conversions panic? 
Maybe we should add another `, err` return value and check it and return error nicely to Rust?

Reply:
Actually I don't familir with the golang. Do you have any suggestions about how to handle the potential error gracefully here? Thank you!

Reply:
See [this example](https://go.dev/tour/methods/15). I think in this case it should be:

```suggestion
	inputOPA, ok := rs[0].Expressions[0].Value.(map[string]interface{})
	if !ok {
	      retrun C.CString("Error:: unexpected type in first expression")
	}
	dataOPA, ok := rs[0].Expressions[1].Value.(map[string]interface{})
	if !ok {
	      retrun C.CString("Error:: unexpected type in second expression")
	}
```

Reply:
Thank you very much!
But it will cause the following error. So I will keep the current implemenation:

  ERROR: # command-line-arguments
  ./opa.go:53:16: cannot assign bool to err (type error) in multiple assignment:
        bool does not implement error (missing Error method)
  ./opa.go:57:15: cannot assign bool to err (type error) in multiple assignment:
        bool does not implement error (missing Error method)
  note: module requires Go 1.16

Reply:
If you've used `err` instead of `ok` as the second return variable this will not work, because `err` is an already-defined variable of type `error` (whereas type assertions return bool as the second return value).